### ISSUE TITLE
Unable to set secrets on a job

### ISSUE THREAD

**Issue by anthonymayer** [Qovery/terraform-provider-qovery#271]
│ Error: Error on job update
│ failed to update job: invalid job upsert request: invalid secrets diff request: invalid secrets upsert request: Key: 'UpsertRequest.Value' Error:Field validation for 'Value'
│ failed on the 'required' tag

I'm trying to set secrets on a new `qovery_job`. I'm setting secrets for the job exactly how I'm setting them for a `qovery_application` but it fails. I tried both when creating the job and when that didn't work I created the job with no secrets and tried to add them as a change and I got the exact same error. Setting environment variables works.

We're running in AWS with the tfstate stored in s3. Let me know if there any any other details you need to reproduce.

**Comment by anthonymayer** [comment]
@benjaminch https://gist.github.com/anthonymayer/fd54ad43a2cf100680dff536db1fabc4 here's the state and the terraform code that sets it up is mostly below. it's a little bit nested, but here are the relevant parts

```
resource "qovery_job" "this" {
  # Required
  environment_id = var.environment_id
  name           = var.name

  # Optional
  cpu    = var.cpu
  memory = var.memory
  environment_variables = [
    for envKey, envValue in var.environment_variables : {
      key   = envKey
      value = envValue
    }
  ]

  secrets = [
    for secKey, secValue in var.secrets : {
      key   = secKey
      value = secValue
    }
  ]

  schedule = var.schedule

  source = {
    docker = {
      dockerfile_path = var.dockerfile_path
      git_repository = {
        url       = var.url
        branch    = var.branch
        root_path = var.root_path
      }
    }
  }

  healthchecks = {  }
}
```
`var.secrets` gets set to `jsondecode(data.aws_secretsmanager_secret_version.insurance_service_secrets.secret_string)`
which comes from
```
data "aws_secretsmanager_secret_version" "insurance_service_secrets" {
  secret_id = "SECRETS_ID_REDACTED"
}
```
we do the exact same thing with `aws_secretsmanager_secret_version` and the loop to set the secrets for `qovery_application` and it works fine there

**Comment by anthonymayer** [comment]
@benjaminch I believe I figured out what the cause of the problem is. I have some secrets I'm trying to set with an empty value. This isn't a problem for `qovery_application` though, but it also fails on `qovery_environment`. Which behavior should it be? It would be nice if `null` just got translated to an empty string, but I think most important would be consistent behavior.

**Comment by mzottolaqovery** [comment]
Hello @anthonymayer :)
Yes you're right, it should be allowed to set an empty value.
We're currently working on enriching the variables by adding alias & overrides support. The fix for this issue will be embedded too.
I'll tell you when it's done

**Comment by mzottolaqovery** [comment]
Hello @anthonymayer,
We just released the `v0.18.5` that should fix your issue.
